gui/src/app/prototypes/AnalysisPyPrototype/AnalysisPyFileEditor.tsx

```diff
@@ -0,0 +1,159 @@
+import { FunctionComponent, useCallback, useMemo, useState } from "react";
+import { PlayArrow } from "@mui/icons-material";
+import { PyodideInterface, loadPyodide } from "pyodide";
+import TextEditor, { ToolbarItem } from "../../FileEditor/TextEditor";
+
+type Props = {
+    fileName: string
+    fileContent: string
+    onSaveContent: (text: string) => void
+    editedFileContent: string
+    setEditedFileContent: (text: string) => void
+    readOnly: boolean
+    width: number
+    height: number
+    outputDiv?: HTMLDivElement | null
+}
+
+const AnalysisPyFileEditor: FunctionComponent<Props> = ({fileName, fileContent, onSaveContent, editedFileContent, setEditedFileContent, readOnly, width, height, outputDiv}) => {
+    const [status, setStatus] = useState<'idle' | 'loading' | 'running' | 'completed' | 'failed'>('idle')
+
+    const loadPyodideInstance = useMemo(() => {
+        let pyodide: PyodideInterface | null = null
```

Comment:
Is there any hope of sharing code between this and the data file? Not just for code organization reasons, but I also noticed that this seems to re-download pyodide if you run `data.py` and then switch and run `analysis.py`

Reply:
We need to make a decision on whether to use the same pyodide instance for data.py and analysis.py. Since they are going to be depending on a different set of packages (analysis.py will potentially be a lot heavier), I suggest that we use two separate instances even if it means we download twice.

Reply:
Having the variables available from the data script (or at the very least, the `data` variable) has some value, I think